src/System.Private.ServiceModel/tests/Scenarios/Client/ExpectedExceptions/ExpectedExceptionTests.cs

```diff
@@ -186,28 +169,28 @@ public static void TimeoutTest_SendTimeout5Seconds()
     [OuterLoop]
     public static void TimeoutTest_SendTimeout0Seconds()
     {
-        bool exceptionThrown = false;
+        BasicHttpBinding binding = new BasicHttpBinding();
+        binding.SendTimeout = TimeSpan.FromMilliseconds(0);
+        ChannelFactory<IWcfService> factory = new ChannelFactory<IWcfService>(binding, new EndpointAddress(Endpoints.HttpBaseAddress_Basic));
+
+        Stopwatch watch = new Stopwatch();
         try
         {
-            BasicHttpBinding binding = new BasicHttpBinding();
-            binding.SendTimeout = TimeSpan.FromMilliseconds(0);
-            ChannelFactory<IWcfService> factory = new ChannelFactory<IWcfService>(binding, new EndpointAddress(Endpoints.HttpBaseAddress_Basic));
-            IWcfService proxy = factory.CreateChannel();
-            proxy.EchoWithTimeout("Hello");
-        }
-        catch (TimeoutException)
-        {
-            exceptionThrown = true;
+            var exception = Assert.Throws<TimeoutException>(() =>
+            {
+                IWcfService proxy = factory.CreateChannel();
+                watch.Start();
+                proxy.EchoWithTimeout("Hello");
+            });
         }
-        catch (Exception ex)
+        finally
         {
-            Assert.True(false, String.Format("Unexpected exception caught: {0}", ex.ToString()));
+            watch.Stop(); 
         }
 
-        if (!exceptionThrown)
-        {
-            Assert.True(false, "Expected TimeoutException was not thrown nor was any other exception thrown.");
-        }
+        // want to assert that this completed in < 0.5 s as an upper bound since the SendTimeout is 0 sec
+        // (usual case is around 1 - 3 ms) 
+        Assert.InRange<long>(watch.ElapsedMilliseconds, 0, 500);
```

Comment:
But then now we're in the business of verifying timeouts in the order of 10s? 

I dunno. In theory, the CI machines should be fairly well-equipped to handle load, and local machines should also be able to handle this. 

Maybe we give 2 seconds leeway... that's probably good enough for verifying a client scenario. @StephenBonikowsky, what are your thoughts on this upper bound value? 

I maintain that we should check this value, but I'm open to loosening the bounds. I do think that a 10s bound is probably a little too much. (Yes, we're having a discussion on choosing a magic number now...)